### ISSUE TITLE
Format of validation `_ERROR_CODES` hard to read

### ISSUE THREAD

**Issue by hayfield** [IATI/iati.core#106]
The current format of `_ERROR_CODES` (a dictionary with blocks of text in) is not nicely scalable and leads to fairly unreadable content.

Suggest changing to a format that is more suited to containing this sort of information, for example YAML.

**Comment by hayfield** [comment]
If reading in from string, the biggest problem is the base errors. The way the relevant types are found is different in each Python 2 and Python 3. This is due to exceptions becoming builtin in Python 3.

To locate the `ValueError` class:

#### Python 2
```python
import exceptions
getattr(exceptions, 'ValueError')

# OR

import sys
getattr(sys.modules['exceptions'], 'ValueError')
```

#### Python 3
```python
import sys
getattr(sys.modules['builtins'], 'ValueError')
```

#### Python 2 and 3
```python
import sys
try:
  getattr(sys.modules['builtins'], 'ValueError')
except KeyError:
  getattr(sys.modules['exceptions'], 'ValueError')
```

**Comment by hayfield** [comment]
How to do multi-line strings in YAML: https://stackoverflow.com/a/21699210

**Comment by hayfield** [comment]
Fixed in #117
